miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/workflow/DefaultWorkflowManager.java

```diff
@@ -80,7 +80,7 @@ protected ProgressStep makeProgressStep(String input, Set<InputType> inputTypes)
       }
     }
 
-    throw new ValidationException(Collections.singletonList(new ValidationError("Could not construct ProgressStep")));
+    throw new ValidationException(Collections.singletonList(new ValidationError("Could not convert input to any expected type.")));
```

Comment:
Message is still not meaningful to a user. Should be something like "No {...} found matching {input}" where {...} is the list of input types. e.g. "No Pool found matching 'asdf'" or "No Sample, Library, or Dilution found matching 'asdf'"

Reply:
:+1: I will add more details.

Reply:
Inserting the "or" in a string might be challenging, so the message now reads something like:
```
'asdf' does not match any of the expected types: (Pool, Integer)
```